### PR TITLE
Enforce OAuth instanceurl usage in packager

### PR DESCRIPTION
**Pull request by rosswbrown**
Field 'instanceurl' is only supported when used with 'authentication'='oauth'.
- add check to packager
- add tests for positive and negative case

All tests pass: `python setup.py test`

There might be a more compact way to write this check.  Please suggest if so, thanks!

### CODE REVIEW

**Review by lukewrites — APPROVED** [review]
I think we could make the check more compact, but at the cost of readability. This is clear…well, _Pythonicly clear_ 😄

File: connector-packager/connector_packager/xsd_validator.py [hunk 1]
```diff
@@ -163,6 +163,21 @@ def validate_file_specific_rules(file_to_test: ConnectorFile, path_to_file: Path
                                                  " already exists. Cannot have multiple fields with the same name.")
 
                     return False
+                if field_name == 'instanceurl':
+                    used_for_oauth = False
+                    for conditions in child.iter("conditions"):
+                        for condition in conditions.iter("condition"):
+                            if 'field' in condition.attrib:
```
Comment (lukewrites) [inline connector-packager/connector_packager/xsd_validator.py:170]: I don't like this any better.
```suggestion
                            for conditions in child.iter("conditions"):
                        for condition in conditions.iter("condition"):
                            if not used_for_oauth:
                                used_for_oauth = ('field' in condition.attrib) & \
                                                    (condition.attrib['field'] == 'authentication') & \
                                                    ('value' in condition.attrib) & \
                                                    (condition.attrib['value'] == 'oauth')
```

**Review by wangxinlei — APPROVED** [review]
